src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -632,27 +641,77 @@ export const Deposit: React.FC<DepositProps> = ({
       contextDispatch({ type: ThorchainSaversDepositActionType.SET_LOADING, payload: true })
       setIsSendMax(percent === 1)
 
+      const _percentageCryptoAmountBaseUnitBeforeTxFees = bnOrZero(balanceCryptoBaseUnit)
+        .times(percent)
+        .toFixed(0)
+
       const _percentageCryptoAmountPrecisionBeforeTxFees = bnOrZero(balanceCryptoPrecision)
         .times(percent)
         .dp(asset.precision)
 
+      const assetAddress = !isToken(assetId)
+        ? zeroAddress
+        : getAddress(fromAssetId(assetId).assetReference)
+
+      const thorchainDepositQuoteQueryKey: GetThorchainSaversDepositQuoteQueryKey = [
+        'thorchainSaversDepositQuote',
+        { asset, amountCryptoBaseUnit: _percentageCryptoAmountPrecisionBeforeTxFees },
+      ]
+
+      let inboundAddressRouter, depositInputData
+
+      if (isEvmChainId(asset.chainId)) {
+        const [thorchainDepositQuote, inboundAddresses] = await Promise.all([
+          queryClient.fetchQuery({
+            queryKey: thorchainDepositQuoteQueryKey,
+            queryFn: () =>
+              fetchThorchainDepositQuote({
+                asset,
+                amountCryptoBaseUnit: toBaseUnit(
+                  _percentageCryptoAmountPrecisionBeforeTxFees,
+                  asset.precision,
+                ),
+              }),
+          }),
+          queryClient.fetchQuery({
+            ...reactQueries.thornode.inboundAddresses(),
+          }),
+        ])
+
+        const inboundAddressData = selectInboundAddressData(inboundAddresses, assetId)
+        const inboundAddress = inboundAddressData?.address
+        inboundAddressRouter = inboundAddressData?.router
+
+        depositInputData =
+          feeAsset?.chainId && isEvmChainId(feeAsset.chainId)
+            ? depositWithExpiry({
+                vault: inboundAddress ?? '',
+                asset: assetAddress,
+                amount: _percentageCryptoAmountBaseUnitBeforeTxFees,
+                memo: thorchainDepositQuote?.memo ?? '',
+                expiry: BigInt(dayjs().add(15, 'minute').unix()),
+              })
+            : undefined
+      }
+
       const estimateFeesQueryEnabled = Boolean(
-        fromAddress && accountId && isUtxoChainId(asset.chainId),
+        fromAddress && accountId && feeAsset?.assetId === assetId,
```

Comment:
Now that this guy is enabled, `isSweepNeededQueryEnabled` needs to explicitly check for UTXO ChainId similar to `isEstimateSweepFeesQueryEnabled`, or else sweep needed query will run for non-UTXOs.

Reply:
Reverted in [c0cca3f](https://github.com/shapeshift/web/pull/7606/commits/c0cca3f5b1d2711ee907ab0deea35f2cbd0dbe79)